docs/DeveloperGuide.adoc

```diff
@@ -239,7 +239,312 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+
+Step 1. The user launches the application for the first time. 'filteredClubBudgets' will be initialised with an empty list
+of all the club budgets in the address book.
+
+Step 2. The user (a club's treasurer) executes `budget c/Computing Club t/200 e/5` command in order to
+submit the data for the calculation of her club's budget. The 'LogicManager' then calls the 'parseCommand' in the
+'AddressBookParser'.
+
+image::BudgetCommand.png[width="800"]
+
+Step 3. The 'AddressBookParser' then returns a new 'BudgetCommandParser', which then parses the command to be executed and
+creates a 'ClubBudgetElements' object called 'club' with the club's name, the expected turnout and the number of events.
+Finally the 'BudgetCommand' is called with 'club' as the parameter.
+
+Step 4. The 'BudgetCommand' checks whether the 'club' is a duplicate and it is not, the 'BudgetCommand' calls the 'addClub'
+command in 'Model' with 'club' as the parameter.
+
+Step 5. 'Model' calls 'addClub' in 'ReadOnlyAddressBook' and indicates that the address book's status has changed.
+
+Step 6. 'ReadOnlyAddressBook' calls the 'addClubBudget' command on an object 'clubs' of the 'UniqueClubsList' class, thus
+adding the required club's data to the address book.
+
+Step 7. Finally a success message is displayed with the details that have been entered by the user.
+
+image::ExecuteBudgetCommand.png[width="800"]
+
+[NOTE]
+In Step 4, had the user entered a club name that already existed in the list of clubs in the address book, then a duplicate
+message would be shown, prompting the user to edit their entered command and try again. Execution of subsequent steps would be stopped until the
+user entered a unique club name
+
+image::DuplicateBudget.png[width="800"]
+
+The image below is the sequence diagram for the functioning of the `budget` command:
+
+image::BudgetCommandSequenceDiagram.png[width="800"]
+
+//end::budget[]
+
 // tag::undoredo[]
+=== Search Pruning Feature
+
+The Search Pruning mechanism is facilitated by the `SearchHistoryManager`.
+Within `SearchHistoryManager` is `searchHistoryStack`, storing `Predicate` objects.
+The `Predicate` objects are used to filter `filteredListPerson` in ModelManager.
+
+A main `SearchHistoryManager` object named `searchHistoryManager` is stored within `ModelManager` and you could access `searchHistoryManager` by calling the
+getter method `getSearchHistoryManager()`.
+
+IMPORTANT: `searchHistoryManger` is only meant for the filtering of
+`Person` objects and Predicates irrelevant to the filtering of `Person` objects could not be stored in it.
+If you want to utilize `SearchHistoryManager` for your own use case, you can initialize a new SearchHistoryManager object
+by calling its default constructor.
+
+==== Current Implementation
+
+The main implementation behind SearchHistoryManager is a Stack Data Structure and the following 3 methods of `SearchHistoryManager` are exposed for your usage +
```

Comment:
Nice inclusion of Stack, a linear data structure to suit the requirements of your feature.